accepted/2025/prune-package-reference-rollout.md

```diff
@@ -0,0 +1,69 @@
+
+This has 2 benefits:
+
+1. Reduces false positives by NuGet Audit, Component Governance and other dependency auditing tools.
+Our data investigations show that up 30% of transitive vulnerabilities are within packages that may be removed by pruning.
+1. Improves performance, by virtue of having smaller dependency graphs and fewer packages downloaded.
+
+For a feature like this, normally we'd need roll it out based on the target framework, but our telemetry shows that up to 30% of transitive vulnerability warnings may be addressed by enabling the pruning feature.
+The list of packages being removed is the exact same that's part of the build time conflict resolution in the .NET SDK. In a way, this moves the conflict resolution to an earlier part of the build.
+The packages that are being pruned carried assemblies that were excluded from the final output of the project by the aforementioned conflict resolution.
+
+- As we are enabling this feature in .NET 10 Preview 1, it gives a very long runway should we eventually decide to go with a more convervative roll-out strategy.
+- If there are any issues, users can easily disable the feature by setting `RestoreEnablePackagePruning` to `false` in their project/props.
+- We will continuously monitor telemetry and feedback around the feature through all the previews.
+
+### Testing & Validation
+
+The feature has been development since November and the implementation has been extensively tested through automation, but we also ensured that our own repos can dogfood without any issues.
+
+We validated the .NET SDK build that it *does not affect the builds negatively of the following repos*:
+
+- NuGet/NuGet.Client
+- dotnet/sdk
+- dotnet/roslyn - <https://github.com/dotnet/roslyn/pull/76898>
+- dotnet/runtime - <https://github.com/dotnet/roslyn/pull/76896>
+- dotnet/aspnetcore - <https://github.com/dotnet/aspnetcore/pull/60025>
+
+## Risks and drawbacks
+
+It may lead to some build issues for if a build has customization such as:
+
+- If a project manually references an assembly or props/targets from a package from the global packages folder. Note that pruning is not allowed for direct dependencies, so this would need to be a transitive package.
+  - Manual assembly coping - at runtime, the platform version will be preferred anyways, but it could lead to the build failing if the package cannot be found on disk anymore.
+  - Test projects that are used for integration testing may have set-ups that manually copy assemblies to the build output folder for the project.
+- Issues with packages lock files - Customers using lock files may see fewer packages downloaded with the newer SDKs and if they're locked mode, they'll see failures. (Packages lock files are used in about 1% of all projects). For customers with lock files, we recommend that they lock down both their tooling and packages.
+
```

Comment:
Customers who expect the output of their project to produce identical results will notice smaller deps files.  This is intentional.

Customers who observe the output of component-detection / component-governance will notice that these components are no longer reported.  This is also intentional.